pro/src/new_components/OfferIndividualForm/utils/setInitialFormValues.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  FORM_DEFAULT_VALUES,
+  IOfferIndividualFormValues,
+} from 'new_components/OfferIndividualForm'
+import { TOffererName } from 'core/Offerers/types'
+import { TOfferIndividualVenue } from 'core/Venue/types'
+
+const setInitialFormValues = (
+  values: IOfferIndividualFormValues,
+  offererNames: TOffererName[],
+  offererId: string | null,
+  venueId: string | null,
+  venueList: TOfferIndividualVenue[]
+): IOfferIndividualFormValues => {
+  const setOffererId = () => {
+    if (offererNames.length === 1) {
+      return offererNames[0].id
+    }
+    if (offererId && offererNames.find(offerer => offerer.id === offererId)) {
+      return offererId
+    }
+    return FORM_DEFAULT_VALUES.offererId
+  }
+
+  const setVenueId = (formOffererId: string) => {
+    if (formOffererId) {
+      const offererVenues = venueList.filter(
+        (venue: TOfferIndividualVenue) =>
+          venue.managingOffererId === formOffererId
+      )
+      if (offererVenues.length === 1) {
+        return offererVenues[0].id
+      }
+    }
+    if (venueId && venueList.find(venue => venue.id === venueId)) {
+      return venueId
+    }
+    return FORM_DEFAULT_VALUES.venueId
+  }
```

Comment:
J'ai peu têtre lu un peu vite mais quel est l'intérêt de créer des fonctions qu'on appelle direct après ?
ça compléxifie un peu non ?

Reply:
Oui je trouve aussi que ça rend la lecture plus difficile, ça force à faire des back and forth qui font un peu perdre la vue d'ensemble

Reply:
J'ai retiré l'utilisation de fonction [ici](https://github.com/pass-culture/pass-culture-main/pull/2227/files#diff-5f43901b37207976339c3f21c128376697c6c28bf2a287ce16a22c501c98a2afR15-R45).